Python 3, and 27 support

jvishnefske opened the pull request:
This is a significant refactor to the project for use on an internal project. I made an attempt at formating the codebase closer to PEP8 formats, and moved it into a python module. This update will break existing code since I renamed the main package from niScope to niscope in order to move all code into a single module, and conform to style guides for modules. I hope it is useful for the community to have python 3, and python 2.7 support. I was able to sucessfully run the 16 unit tests on a simulated National Instruments PCI-5114 on windows 7.

jvishnefske committed:
added support for 64 bit module

jvishnefske committed:
futurized, and autopep8 converted tabs to spaces.

jvishnefske committed:
manual python 3 conversion

jvishnefske committed:
created module

jvishnefske committed:
Create LICENSE

jvishnefske committed:
Delete LICENSE

bernardokyotoku commented:
Wow, really awesome.

John, can I transfer the github repository ownership to you? I am not in
the field anymore. I don't even have access to access to a board that runs
niscope. I thinks this link is nice because it is referenced by national
instruments.

Cheers,

Bernardo

2017-06-16 14:34 GMT-03:00 John Vishnefske <notifications@github.com>:

> This is a significant refactor to the project for use on an internal
> project. I made an attempt at formating the codebase closer to PEP8
> formats, and moved it into a python module. This update will break existing
> code since I renamed the main package from niScope to niscope in order to
> move all code into a single module, and conform to style guides for
> modules. I hope it is useful for the community to have python 3, and python
> 2.7 support. I was able to sucessfully run the 16 unit tests on a simulated
> National Instruments PCI-5114 on windows 7.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/bernardokyotoku/pyniscope/pull/4
> Commit Summary
>
>    - added support for 64 bit module
>    - futurized, and autopep8 converted tabs to spaces.
>    - manual python 3 conversion
>    - created module
>    - Create LICENSE
>    - Delete LICENSE
>
> File Changes
>
>    - *M* .gitignore
>    <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-0>
>    (73)
>    - *D* niScope.py
>    <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-1>
>    (579)
>    - *D* niScopeTypes.py
>    <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-2>
>    (521)
>    - *A* niscope/__init__.py
>    <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-3>
>    (587)
>    - *A* niscope/niScopeTypes.py
>    <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-4>
>    (973)
>    - *R* niscope/ordered_symbols.py
>    <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-5>
>    (1679)
>    - *A* patch
>    <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-6>
>    (2303)
>    - *M* setup.py
>    <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-7>
>    (10)
>    - *M* test/ExportSignalActive.py
>    <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-8>
>    (15)
>    - *M* test/test_niscope.py
>    <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-9>
>    (199)
>    - *A* tox.ini
>    <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-10>
>    (5)
>
> Patch Links:
>
>    - https://github.com/bernardokyotoku/pyniscope/pull/4.patch
>    - https://github.com/bernardokyotoku/pyniscope/pull/4.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bernardokyotoku/pyniscope/pull/4>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAgGYGy9EizCr0hssrhufWRyyENbSCo2ks5sEryegaJpZM4N8u6x>
> .
>

jvishnefske commented:
That would be great Bernardo. I am using if for a consulting job, and may
put it on pypi at some point.

On Fri, Jun 16, 2017 at 1:33 PM, Bernardo Kyotoku <notifications@github.com>
wrote:

> Wow, really awesome.
>
> John, can I transfer the github repository ownership to you? I am not in
> the field anymore. I don't even have access to access to a board that runs
> niscope. I thinks this link is nice because it is referenced by national
> instruments.
>
> Cheers,
>
> Bernardo
>
> 2017-06-16 14:34 GMT-03:00 John Vishnefske <notifications@github.com>:
>
> > This is a significant refactor to the project for use on an internal
> > project. I made an attempt at formating the codebase closer to PEP8
> > formats, and moved it into a python module. This update will break
> existing
> > code since I renamed the main package from niScope to niscope in order to
> > move all code into a single module, and conform to style guides for
> > modules. I hope it is useful for the community to have python 3, and
> python
> > 2.7 support. I was able to sucessfully run the 16 unit tests on a
> simulated
> > National Instruments PCI-5114 on windows 7.
> > ------------------------------
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/bernardokyotoku/pyniscope/pull/4
> > Commit Summary
> >
> > - added support for 64 bit module
> > - futurized, and autopep8 converted tabs to spaces.
> > - manual python 3 conversion
> > - created module
> > - Create LICENSE
> > - Delete LICENSE
> >
> > File Changes
> >
> > - *M* .gitignore
> > <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-0>
> > (73)
> > - *D* niScope.py
> > <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-1>
> > (579)
> > - *D* niScopeTypes.py
> > <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-2>
> > (521)
> > - *A* niscope/__init__.py
> > <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-3>
> > (587)
> > - *A* niscope/niScopeTypes.py
> > <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-4>
> > (973)
> > - *R* niscope/ordered_symbols.py
> > <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-5>
> > (1679)
> > - *A* patch
> > <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-6>
> > (2303)
> > - *M* setup.py
> > <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-7>
> > (10)
> > - *M* test/ExportSignalActive.py
> > <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-8>
> > (15)
> > - *M* test/test_niscope.py
> > <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-9>
> > (199)
> > - *A* tox.ini
> > <https://github.com/bernardokyotoku/pyniscope/pull/4/files#diff-10>
> > (5)
> >
> > Patch Links:
> >
> > - https://github.com/bernardokyotoku/pyniscope/pull/4.patch
> > - https://github.com/bernardokyotoku/pyniscope/pull/4.diff
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/bernardokyotoku/pyniscope/pull/4>, or mute the
> thread
> > <https://github.com/notifications/unsubscribe-auth/
> AAgGYGy9EizCr0hssrhufWRyyENbSCo2ks5sEryegaJpZM4N8u6x>
>
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bernardokyotoku/pyniscope/pull/4#issuecomment-309101808>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AILUfcEzNhWZ6gxmERhttjxWjjIHD9SIks5sEsqKgaJpZM4N8u6x>
> .
>

marcoskirsch commented:
Hello,

I'm a Software Engineer at National Instruments in the Modular Instruments group. On the last [NIWeek](http://niweek.ni.com/events/niweek-2017/) – NI's yearly convention, we announced Python support for several of our drivers, including NI-SCOPE.

We are now actively working on this. Specifically, for NI-SCOPE and other Modular Instruments, we are writing the Python bindings as an Open Source project hosted on GitHub: [nimi-python](https://github.com/ni/nimi-python). We have committed to having five hardware drivers available on PyPI by the end of the year. In fact, you can already get an early version of [nidmm](https://pypi.python.org/pypi/nidmm/0.1.0) and utility API [nimodinst](https://pypi.python.org/pypi/nimodinst/0.1.0).

As such, we'd like to invite you to use [nimi-python](https://github.com/ni/nimi-python). We're using code-generation in order to ensure consistency, completeness, and maintainability across Modular Instruments APIs. Given this, it was not feasible for us to contribute to existing hand-coded wrappers such as this.

I can speak for myself and my colleagues working on this project when I say that we're excited about Python and the applications customers will build using it along with our hardware. We welcome your feedback.
